controllers/openstackcluster_controller.go

```diff
@@ -308,6 +318,27 @@ func reconcileNormal(scope scope.Scope, cluster *clusterv1.Cluster, openStackClu
 	return reconcile.Result{}, nil
 }
 
+func getOrCreateBastionPorts(scope scope.Scope, clusterName string, openStackCluster *infrav1.OpenStackCluster, machinePorts []infrav1.PortOpts, trunkEnabled bool, securityGroups []infrav1.SecurityGroupFilter, instanceName string, networkingService *networking.Service) ([]infrav1.PortStatus, error) {
+	var portsStatus []infrav1.PortStatus
+	var err error
+
+	if openStackCluster.Status.Bastion == nil || (openStackCluster.Status.Bastion != nil && openStackCluster.Status.Bastion.PortsStatus == nil) {
```

Comment:
The Bastion will be recreated if `instanceSpec` hash has changed so yes if the user updates `OpenStackCluster.Spec.Bastion.Instance.Ports`, because of that conditional the ports will not be removed. I need to fix this.

Reply:
nice catch, no I haven't addressed this yet. I'll do it asap.

Reply:
in fact I think it does, see https://github.com/shiftstack/cluster-api-provider-openstack/blob/05b31bbca8872f00647c98c217fddc4038df4166/controllers/openstackcluster_controller.go#L419-L422
We delete the bastion if the hash has changed and in this function we remove the associated ports: https://github.com/shiftstack/cluster-api-provider-openstack/blob/05b31bbca8872f00647c98c217fddc4038df4166/controllers/openstackcluster_controller.go#L252-L262
and set Status fields to `nil`.

So I think we effectively cleanup stuffs in the right order when a bastion needs re-creation.